base/special/gamma.jl

```diff
@@ -482,3 +482,7 @@ eta(x::Integer) = eta(Float64(x))
 eta(x::Real)    = oftype(float(x),eta(Float64(x)))
 eta(z::Complex) = oftype(float(z),eta(Complex128(z)))
 @vectorize_1arg Number eta
+
+Base.binomial{S<:Number,T<:Number}(n::S, k::T) = ( (isinteger(n) && isinteger(k))
```

Comment:
ah, right. the generic `isinteger` for floating point numbers is comparing whether the value is equal to `trunc` of itself and is finite, which doesn't mean it'll necessarily fit in an `Int64` - for type stability I wonder whether it would be any simpler to just always use the definition in terms of `gamma`